library/modules/bigip_snat_translation.py

```diff
@@ -0,0 +1,694 @@
+        'udpIdleTimeout': 'udp_idle_timeout'
+    }
+
+    api_attributes = [
+        'address',
+        'arp',
+        'connectionLimit',
+        'description',
+        'disabled',
+        'enabled',
+        'ipIdleTimeout',
+        'tcpIdleTimeout',
+        'trafficGroup',
+        'udpIdleTimeout'
+    ]
+
+    returnables = [
+        'address',
+        'arp',
+        'connection_limit',
+        'description',
+        'disabled',
+        'enabled',
+        'ip_idle_timeout',
+        'state',
+        'tcp_idle_timeout',
+        'traffic_group',
+        'udp_idle_timeout'
+    ]
+
+    updatables = [
+        'arp',
+        'connection_limit',
+        'description',
+        'disabled',
+        'enabled',
+        'ip_idle_timeout',
+        'traffic_group',
+        'tcp_idle_timeout',
+        'udp_idle_timeout'
+    ]
+
+    @property
+    def traffic_group(self):
+        if self._values['traffic_group'] is None:
+            return None
+        else:
+            result = fq_name(self.partition, self._values['traffic_group'])
+
+
+class ApiParameters(Parameters):
+    @property
+    def connection_limit(self):
+        if self._values['connection_limit'] is None:
+            return None
+        return int(self._values['connection_limit'])
+
+    @property
+    def enabled(self):
+        if 'enabled' in self._values:
+            return True
+        return False
+
+    @property
+    def disabled(self):
+        if 'disabled' in self._values:
+            return True
+        return False
+
+
+class ModuleParameters(Parameters):
+    def _clear_member_prefix(self, address):
+        result = os.path.basename(address)
+        return result
+
+    def _format_ip_address(self, address):
+        if is_valid_ip(address):
+            address = '/{0}/{1}'.format(self.partition, address)
+            return address
+        else:
+            raise F5ModuleError(
+                'The provided member address is not a valid IP address'
+            )
+
+    @property
+    def arp(self):
+        result = flatten_boolean(self._values['arp'])
+        if result is None:
+            return None
+        if result == 'yes':
+            return 'enabled'
+        return 'disabled'
+
+    @property
+    def connection_limit(self):
+        if self._values['connection_limit'] is None:
+            return None
+        return int(self._values['connection_limit'])
+
+    @property
+    def description(self):
+        if self._values['description'] in [None, 'none']:
+            return None
+        return self._values['description']
+
+    @property
+    def enabled(self):
+        if self._values['state'] in ['enabled', 'present']:
+            return True
+
+    @property
+    def state(self):
+        if self.enabled is True and self._values['state'] != 'present':
+            return 'enabled'
+        elif self.disabled is True:
+            return 'disabled'
+        else:
+            return self._values['state']
+
```

Comment:
You are missing integer validation for:  connection_limit, ip_idle_timeout, udp_idle_timeout, udp_idle_timeout, connection_limit

something similar to this:

https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_firewall_log_profile.py#L369